test/cql-pytest/test_describe.py

```diff
@@ -733,6 +733,49 @@ def test_table_options_quoting(cql, test_keyspace):
     finally:
         cql.execute(f"DROP TYPE {test_keyspace}.\"{type_name}\"")
 
+@pytest.mark.parametrize("test_keyspace",
+                         [pytest.param("tablets", marks=[pytest.mark.xfail(reason="issue #16317")]), "vnodes"],
+                         indirect=True)
+def test_hide_cdc_table(scylla_only, cql, test_keyspace):
+    cdc_table_suffix = "_scylla_cdc_log"
+    
+    with new_test_table(cql, test_keyspace, "a int primary key, b int", "WITH cdc = {'enabled': true}") as t:
+        t_name = t.split('.')[1]
+        cdc_log_name = t_name + cdc_table_suffix
+
+        # Check 't_scylla_cdc_log' is hidden in DESC TABLES
+        desc_tables = cql.execute("DESC TABLES")
+        assert cdc_log_name not in [r.name for r in desc_tables]
+        cdc_log_table_entry = cql.execute(f"SELECT * FROM system_schema.tables WHERE keyspace_name='{test_keyspace}' AND table_name='{cdc_log_name}'").all()
+        assert len(cdc_log_table_entry) == 1 # assert the table exists
+
+        # Check 't_scylla_cdc_log' is hidden in DESC KEYSPACE ks
+        desc_keyspace = cql.execute(f"DESC KEYSPACE {test_keyspace}").all()
+        for row in desc_keyspace:
+            assert row.name != cdc_log_name or (row.name == cdc_log_name and f"ALTER TABLE {test_keyspace}.{cdc_log_name} WITH" in row.create_statement)
+
+        # Check 't_scylla_cdc_log' is hidden in DESC SCHEMA
+        desc_schema = cql.execute("DESC SCHEMA")
+        for row in desc_schema:
+            assert row.name != cdc_log_name or (row.name == cdc_log_name and f"ALTER TABLE {test_keyspace}.{cdc_log_name} WITH" in row.create_statement)
+
+        # Check 't_scylla_cdc_log' cannot be described directly
+        with pytest.raises(InvalidRequest, match=f"CDC log table {test_keyspace}.{cdc_log_name} cannot be described."):
+            desc_cdc_table = cql.execute(f"DESC TABLE {test_keyspace}.{cdc_log_name}")
+        
+        # Check base table description contains ALTER TABLE statement for cdc log table
+        desc_base_table = cql.execute(f"DESC TABLE {t}").all()
+        assert f"ALTER TABLE {test_keyspace}.{cdc_log_name} WITH" in desc_base_table[1].create_statement
+        
+        cql.execute(f"DROP TABLE {t}")
```

Comment:
Is it OK to drop the table whose lifetime is managed by `new_test_table`? What will happen if dropped the table but didn't recreate it later - e.g. we got an exception before we got to CREATE TABLE or there was no CREATE TABLE in the `desc_keyspace` output due to a bug?

Reply:
It's ok to drop the table and immediately recreate it (as I believe he does in this case). If the code dropped a table created by new_test_table() and left it deleted (which I believe isn't the case here), the new_test_table() cleanup will throw an exception, when it will run DROP TABLE and fail.